Merge lap-leaseup into master #164876979

jrubenoff opened the pull request:
@chrisdolendo The `lap-leaseup` branch has some merge conflicts. In order to resolve them, I forked it and merged `master` into this branch. Once this PR is merged, we can delete both branches.

---

It looks like the only webapp components this PR touches are:

- The full-page loading spinner
- The content cards on the Additional Resources page

To help you test them, I added [review instructions](https://www.pivotaltracker.com/n/projects/1405352/stories/164876979) to the Pivotal ticket.

Let me know if you have any questions!

chrisdolendo commented:
<img width="1280" alt="Screen Shot 2019-04-02 at 10 47 04 AM (2)" src="https://user-images.githubusercontent.com/5904673/55424251-ae279200-5534-11e9-9bff-9b87f7a6164e.png">

Web app looks good, LAP has this issue I'm seeing when grunting.

jrubenoff commented:
Hmm, I guess I assumed that LAP had the latest styles from the `lap-leaseup` branch, but it doesn't. So your screenshot actually points out a few separate bugs. 

I'll try to fix this as soon as I can, hopefully today but possibly later in the week.

jrubenoff commented:
@chrisdolendo Alrighty, I think this is now in a good place.

### Some background

The PL currently compiles three stylesheets:

- `toolkit.scss` (which `web` uses)
- `toolkit-lap.scss` (which we used for Partners, to avoid affecting webapp styles during its development)
- `toolkit-base.scss` (which seems vestigial, since we don't reference it anywhere)

The thing is, Lease Up dev is done, and the PL has almost no app-specific styles in the PL (except for the webapp's sticky footer.)

So, using all these different stylesheets seems overly complex, and I can't see a reason not to use `toolkit` everywhere.

### List of changes

- Removed `toolkit-base` and `toolkit-lap`
- Moved all Partners styles into `toolkit`
- Changed the sticky footer styles (our only app-specific CSS) to apply only to Web
- Restored / fixed some styles for `.loading` and `.ReactTable`, which I'd changed while resolving merge conflicts

### How to test

I've opened Exygy/sf-dahlia-lap#211 and Exygy/sf-dahlia-web#1182 to update our PL styles with what's in this branch.

You can click through them to confirm:

- The loading overlays look OK
- The issues in the above screenshot have been fixed

If both are true, I think we're good to merge all three PRs at once.

### A note

For any pattern library PRs going forward, it seems like we should open LAP and Web PRs alongside them. This will keep everything up to date, and prevent any future CSS devs from having to debug months-old style changes.

I know this represents some annoying overhead. In the future, it seems like we can eliminate this by turning the PL into an NPM package or Ruby gem. This will let us manage PL versions, plan style updates, etc.

I was talking to @slowbot on Monday as he prepares to work on regional DAHLIA projects, and this sounds like the road he's planning to go down.
